tangelo/toolboxes/post_processing/histogram.py

```diff
@@ -0,0 +1,209 @@
+            outcome can be an int (number of replicas) or a float (probability).
+        n_shots (int): Self-explanatory. If it is greater than 0, the class
+            considers that probabilities are provided.
+        msq_first (bool): Bit ordering. For example, 011 (msq_first) = 110
+            (lsq_first). This depends on the hardware provider convention.
+        epsilon (float): When probabilities are provided, this parameter is used
+            to check if the sum is within [1-epsilon, 1+epsilon]. Default is
+            1e-2.
+    """
+
+    def __init__(self, outcomes, n_shots=0, msq_first=False, epsilon=1e-2):
+
+        # Error: bitstrings not of consistent length.
+        lengths_bitstrings = set([len(k) for k in outcomes.keys()])
+        if len(lengths_bitstrings) > 1:
+            raise ValueError(f"Entries in outcomes dictionary not of consistent length. Please check your input.")
+
+        if n_shots > 0:
+            # Flags histograms whose frequencies do not add to 1.
+            sum_values = sum(outcomes.values())
+            if abs(sum_values-1) > epsilon:
+                raise ValueError(f"Histogram frequencies not summing very close to 1. (sum = {sum_values}).")
+
+            outcomes = {k: round(v*n_shots) for k, v in outcomes.items()}
+
+        self.counts = outcomes.copy()
+
+        if msq_first:
+            self.counts = {k[::-1]: v for k, v in self.counts.items()}
+
+    def __repr__(self):
+        """Output the string representation."""
+        return f"{self.counts}"
+
+    def __eq__(self, other):
+        """Two histograms are equal if they have same counts, i.e. same
+        bistrings and outcomes.
+        """
+        return (self.counts == other.counts)
+
+    def __add__(self, other):
+        """The aggregate_histograms function is used to add two Histogram objects."""
+        return aggregate_histograms(self, other)
+
+    def __iadd__(self, other):
+        new_histogram = self + other
+        self.__dict__ = new_histogram.__dict__
+        return self
+
+    @property
+    def n_shots(self):
+        """Return the number of shots encoded in the Histogram.
+
+        Returns:
+            int: Self-explanatory.
+        """
+        return sum(self.counts.values())
+
+    @property
+    def n_qubits(self):
+        """The length of the bistrings represents the number of qubits."
+
+        Returns:
+            int: Self-explanatory.
+        """
+        return len(list(self.counts.keys())[0])
+
+    @property
+    def frequencies(self):
+        """The frequencies are normalized counts vs the number of shots.
+
+        Returns:
+            dict: Frequencies in a {bistring: probability, ...} format.
+        """
+        return {bistring: counts/self.n_shots for bistring, counts in self.counts.items()}
+
+    def post_select(self,  expected_outcomes):
+        """Post selection is done with the post_select function (see the
```

Comment:
How about we say "Method to apply post selection on Histogram data, based on a post selection function in this module." for the first part